File: compiler/go/IDL/ClassDefinition.go
package IDL

//--------------------------------------------------------------------
type ClassDefinition struct {
	Name         string                   `json:"name"`
	Comment      string                   `json:"comment"`
	Tags         map[string]string        `json:"tags"`
	FunctionPath map[string]string        `json:"function_path"`
	Constructors []*ConstructorDefinition `json:"constructors"`
	Releaser     *ReleaseDefinition       `json:"release"`
	Methods      []*MethodDefinition      `json:"methods"`
	Fields       []*FieldDefinition       `json:"fields"`
}

//--------------------------------------------------------------------
func NewClassDefinition(name string) *ClassDefinition {
	c := &ClassDefinition{
		Name:         name,
		Comment:      "",
		Tags:         make(map[string]string),
		Constructors: make([]*ConstructorDefinition, 0),
		Releaser:     nil,
		Methods:      make([]*MethodDefinition, 0),
		Fields:       make([]*FieldDefinition, 0),
	}
	
	c.Releaser = NewReleaserDefinition(c, "Release"+c.Name)
	
	return c
}

//--------------------------------------------------------------------
func (this *ClassDefinition) SetTag(tag string, val string) {
	
	if this.Tags == nil {
		this.Tags = make(map[string]string)
	}
	
	this.Tags[tag] = val
}

//--------------------------------------------------------------------
func (this *ClassDefinition) AppendComment(comment string) {
	if this.Comment != "" {
		this.Comment += "\n"
	}
	
	this.Comment += comment
}

//--------------------------------------------------------------------
func (this *ClassDefinition) AddConstructor(definition *ConstructorDefinition) {
	definition.SetParent(this)
	this.Constructors = append(this.Constructors, definition)
}

//--------------------------------------------------------------------
func (this *ClassDefinition) AddMethod(def *MethodDefinition) {
	def.SetParent(this)
	this.Methods = append(this.Methods, def)
}

//--------------------------------------------------------------------
func (this *ClassDefinition) SetFunctionPath(key string, val string) {
	
	if this.FunctionPath == nil {
		this.FunctionPath = make(map[string]string)
	}
	
	this.FunctionPath[key] = val
	
	for _, f := range this.Fields {
		if f.Getter != nil {
			f.Getter.SetFunctionPath(key, val)
		}
		if f.Setter != nil {
			f.Setter.SetFunctionPath(key, val)
		}
	}
	
	for _, cstr := range this.Constructors {
		cstr.SetFunctionPath(key, val)
	}
	
	if this.Releaser != nil {
		this.Releaser.SetFunctionPath(key, val)
	}
	
	for _, m := range this.Methods {
		m.SetFunctionPath(key, val)
	}
}

//--------------------------------------------------------------------
func (this *ClassDefinition) AddField(f *FieldDefinition) {
	this.Fields = append(this.Fields, f)
}

//--------------------------------------------------------------------

File: idl_entities/jvm/src/com/metaffi/idl/entities/ClassDefinition.java
package com.metaffi.idl.entities;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ClassDefinition {
    private String name;
    private String comment;
    private Map<String, String> tags;
    private Map<String, String> entity_path;
    private List<ConstructorDefinition> constructors;
    private ReleaseDefinition release;
    private List<MethodDefinition> methods;
    private List<FieldDefinition> fields;

    public ClassDefinition() {
        this.tags = new HashMap<>();
        this.entity_path = new HashMap<>();
        this.constructors = new ArrayList<>();
        this.methods = new ArrayList<>();
        this.fields = new ArrayList<>();
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public Map<String, String> getTags() {
        return tags;
    }

    public void setTags(Map<String, String> tags) {
        this.tags = tags;
    }

    public Map<String, String> getEntityPath() {
        return entity_path;
    }

    public void setEntityPath(Map<String, String> entityPath) {
        this.entity_path = entityPath;
    }

    public List<ConstructorDefinition> getConstructors() {
        return constructors;
    }

    public void setConstructors(List<ConstructorDefinition> constructors) {
        this.constructors = constructors;
    }

    public ReleaseDefinition getRelease() {
        return release;
    }

    public void setRelease(ReleaseDefinition release) {
        this.release = release;
    }

    public List<MethodDefinition> getMethods() {
        return methods;
    }

    public void setMethods(List<MethodDefinition> methods) {
        this.methods = methods;
    }

    public List<FieldDefinition> getFields() {
        return fields;
    }

    public void setFields(List<FieldDefinition> fields) {
        this.fields = fields;
    }
}
